### PR TITLE
feat(ui): reduce number of retries after proxy restart

### PR DESCRIPTION
**Pull request by geigerzaehler**
With #1153 the proxy restarts much quicker after the keystore is unsealed or created. This allows us to reduce the number of retries and fail earlier if the proxy does not start properly. (We now wait 5
seconds).

As part of this change we also adjust the signature of `withRetry`. Now the caller is able to control all timing related parameters.

### CODE REVIEW

**Review by FintanH — DISMISSED** [review]
:clock1:

**Review by xla — DISMISSED** [review]
💋 🕜 📚 🍗

File: ui/src/api.ts [hunk 1]
```diff
@@ -119,13 +119,14 @@ const delay = (delay: number) => {
...
+  delayTime: number,
+  retries: number
 ): Promise<T> => {
-  for (let retries = 0; ; retries++) {
+  for (; ; retries--) {
     try {
       return await request();
     } catch (error) {
-      if (error.message !== "Failed to fetch" || retries > 200) {
+      if (error.message !== "Failed to fetch" || retries > 0) {
```
Comment (MeBrei) [inline ui/src/api.ts:129]:
```suggestion
      if (error.message !== "Failed to fetch" || retries < 0) {
```

Or am I understanding the loop wrong?

**Review by MeBrei — CHANGES_REQUESTED** [review]

File: ui/src/api.ts (hunk 1 above)
Comment (geigerzaehler) [reply ui/src/api.ts]: Great catch!

**Review by MeBrei — APPROVED** [review]

**Review by FintanH — APPROVED** [review]
Nice catch @MeBrei!

**Review by rudolfs — APPROVED** [review]

**Pull request merged**